File: solid-kata/java/lsp/src/test/java/com/codurance/lsp/FillingStationShould.java
package com.codurance.lsp;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.catchThrowable;

class FillingStationShould {
    private static final int FULL = 100;
    private final FillingStation fillingStation = new FillingStation();

    @Test
    void refuel_a_petrol_car() {
        PetrolCar car = new PetrolCar();

        fillingStation.refuel(car);

        assertThat(car.fuelTankLevel())
                .isEqualTo(FULL);
    }


    @Test
    void not_fail_refueling_an_electric_car() {
        ElectricCar car = new ElectricCar();

        Throwable throwable = catchThrowable(() -> fillingStation.refuel(car));

        assertThat(throwable)
                .isNull();
    }


    @Test
    void recharge_an_electric_car() {
        ElectricCar car = new ElectricCar();

        fillingStation.charge(car);

        assertThat(car.batteryLevel())
                .isEqualTo(FULL);
    }


    @Test
    void not_fail_recharging_a_petrol_car() {
        PetrolCar car = new PetrolCar();

        Throwable throwable = catchThrowable(() -> fillingStation.charge(car));

        assertThat(throwable)
                .isNull();
    }
}

File: solid-kata/c#/LSP.Tests/FillingStationShould.cs
using FluentAssertions;
using Xunit;

namespace LSP.Tests
{
    public class FillingStationShould
    {
        private const int Full = 100;
        private readonly FillingStation _fillingStation = new();

        [Fact]
        public void Refuel_A_Petrol_Car()
        {
            var car = new PetrolCar();

            _fillingStation.Refuel(car);

            car.FuelTankLevel
                .Should()
                .Be(Full);
        }

        [Fact]
        public void Not_Fail_Refueling_An_Electric_Car() =>
            _fillingStation
                .Invoking(_ => _.Refuel(new ElectricCar()))
                .Should()
                .NotThrow();

        [Fact]
        public void Recharge_An_Electric_Car()
        {
            var car = new ElectricCar();

            _fillingStation.Charge(car);

            car.BatteryLevel
                .Should()
                .Be(Full);
        }

        [Fact]
        public void Not_Fail_Recharging_A_Petrol_Car() =>
            _fillingStation
                .Invoking(_ => _.Charge(new PetrolCar()))
                .Should()
                .NotThrow();
    }
}